include/exadg/solvers_and_preconditioners/preconditioners/block_jacobi_preconditioner.h

```diff
@@ -36,7 +36,7 @@ class BlockJacobiPreconditioner : public PreconditionerBase<typename Operator::v
     : underlying_operator(underlying_operator_in)
   {
     // initialize block Jacobi
-    underlying_operator.update_block_diagonal_preconditioner();
+    underlying_operator.initialize_block_diagonal_preconditioner();
```

Comment:
I think this is the important line of code related to this comment https://github.com/exadg/exadg/pull/602#discussion_r1398171895. I thought it is an advantage to get rid of the variable `OperatorBase::block_jacobi_preconditioner_is_initialized`.

In the long run, the responsibility should be shifted even more strongly to the `BlockJacobiPreconditioner` (the relevant data structures currently live in `OperatorBase`). I discussed this a lot with @bergbauer, but the solution is not that clear/easy and we still have to work on this.

Reply:
Yes I have seen this. I think the current behavior is fine, I was just thinking about #602, where we aim to trigger updates in the solver. There, one motivation was to not be dependent on construction sequences of solver/preconditioner/operator etc to have all the scaling factors for mass and stiffness matrices avaible when needed. This is just something we should have in the back of our heads here ... probably it just belongs to  that separate PR #602 .

Reply:
Agree, but - for clarifty - let me mention it should be PR #599 where this is done.